Makefile

```diff
@@ -129,3 +131,16 @@ local_gems_bundle:
 
 local_gems_run: local_gems_bundle
 	BUNDLE_GEMFILE=Gemfile-dev make run
+
+build_artifact:
+	bundle config set --local cache_all true
+	bundle package
+	tar --exclude './config/agencies.yml' --exclude './config/iaa_gtcs.yml' \
+		--exclude './config/iaa_orders.yml' --exclude './config/iaa_statuses.yml' \
+		--exclude './config/integration_statuses.yml' --exclude './config/integrations.yml' \
+		--exclude './config/partner_account_statuses.yml' --exclude './config/partner_accounts.yml' \
+		--exclude './config/service_providers.yml' --exclude='./certs/sp' \
+		--exclude='./identity-idp-config' --exclude='./tmp' --exclude='./node_modules' \
+		--exclude='./geo_data/GeoLite2-City.mmdb' --exclude='./pwned_passwords/pwned_passwords.txt' \
+		--exclude='./vendor/ruby' \
+		--exclude='./config/application.yml' -cf - $(ARTIFACT_DIRECTORY) | $(GZIP_COMMAND) > $(ARTIFACT_DESTINATION_FILE)
```

Comment:
do we need a default value for `ARTIFACT_DIRECTORY`?

Reply:
It's hard to have a good default because the command may be run from within the directory or from a directory up, and we don't want to generate the file within the directory being compressed.

Reply:
what if we used `./tmp` since that's excluded already?

Reply:
`ARTIFACT_DIRECTORY` is the source directory being compressed that contains the IDP, which is something I'd rather have explicitly provided by the caller. I considered also making `ARTIFACT_DESTINATION_FILE` not have a default for the same reasons.

I've added a comment to clarify `ARTIFACT_DIRECTORY`.

Reply:
Decided to drop the default for `ARTIFACT_DESTINATION_FILE` in 94d0888cb